Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1793,19 +1793,24 @@ open class NavigationMapView: UIView {
                    routesPresentationStyle: RoutesPresentationStyle = .all(),
                    animated: Bool = false) {
         let geometry: Geometry
+        let customCameraOptions: MapboxMaps.CameraOptions?
         
-        if case let RoutesPresentationStyle.all(shouldFit) = routesPresentationStyle, shouldFit {
-            geometry = .multiLineString(MultiLineString(routes.map({ $0.shape?.coordinates ?? [] })))
-        } else {
+        switch routesPresentationStyle {
+        case .single(cameraOptions: let cameraOptions):
             geometry = .lineString(LineString(routes.first?.shape?.coordinates ?? []))
+            customCameraOptions = cameraOptions
+        case .all(shouldFit: let shouldFit, cameraOptions: let cameraOptions):
+            geometry = shouldFit ? .multiLineString(MultiLineString(routes.map({ $0.shape?.coordinates ?? [] }))) : .lineString(LineString(routes.first?.shape?.coordinates ?? []))
```

Comment:
Small nit: Ternary operators are a bit hard to read here, so I'd probably just use usual if/else.